tests/spark/readers/test_memory.py

```diff
@@ -14,10 +14,10 @@ class TestInMemoryDataReader:
         "data,format,params,expect_filter",
         [
             pytest.param(
-                "id,string\n1,hello,\n2,world", DataFormat.CSV, {"header": True}, "id < 3"
+                "id,string\n1,hello\n2,world", DataFormat.CSV, {"header":0}, "id < 3"
             ),
             pytest.param(
-                b"id,string\n1,hello,\n2,world", DataFormat.CSV, {"header": True}, "id < 3"
+                b"id,string\n1,hello\n2,world", DataFormat.CSV, {"header":0}, "id < 3"
             ),
```

Comment:
This seems like we changed our API (although perhaps in a very minor/subtle way). Let's verify if we shouldn't put backward compatibility in place here.

Reply:
Fixed, put some code for backward compatibility